Rakefile

```diff
@@ -44,72 +101,70 @@ BASH
           else
             curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v#{protoc_version}/protoc-#{protoc_version}-linux-x86_64.zip
           fi
-          unzip protoc-#{protoc_version}*.zip
+
+          mkdir -p #{toolchain_bin_dir}
+          unzip -o protoc-#{protoc_version}*.zip
           mv bin/protoc #{protoc_binary}
-        fi
+
+          mkdir -p #{toolchain_include_dir}
+          echo "here"
+          rm -rf #{toolchain_include_dir}/**
+          mv include/* #{toolchain_include_dir}
+         fi
 BASH
-      EOF
-    end
-  end
+       EOF
+     end
+   end
+
+  task :protoc => ['install_protoc', 'setup_gogo'] do
 
-  task :protoc => ['install_protoc'] do
     sh <<-EOF
       /bin/bash <<BASH
       set -euo pipefail
 
       export GO111MODULE=auto
 
-      rm -rf #{gogo_dir}
-      rm -rf /tmp/gogo-bin-*
-
-      mkdir -p #{gogo_dir}/src/github.com/gogo
-      git clone https://github.com/gogo/protobuf.git #{gogo_dir}/src/github.com/gogo/protobuf
-
-      # Install v1.3.2
-      pushd #{gogo_dir}/src/github.com/gogo/protobuf
-      git checkout v1.3.2
-      GOBIN=/tmp/gogo-bin-v1.3.2 GOPATH=#{gogo_dir} make clean install
-
-      popd
-
       echo "Generating logs proto"
-      PATH=/tmp/gogo-bin-v1.3.2 #{protoc_legacy_binary} --proto_path=$GOPATH/src:#{gogo_dir}/src:. --gogofast_out=$GOPATH/src --java_out=java proto/logs/agent_logs_payload.proto
+      PATH=#{toolchain_bin_dir} #{protoc_legacy_binary} --proto_path=$GOPATH/src:#{gogo_dir}/src:. --gogofast_out=$GOPATH/src --java_out=java proto/logs/agent_logs_payload.proto
 
       echo "Generating metrics proto (go)"
-      PATH=/tmp/gogo-bin-v1.3.2 #{protoc_legacy_binary} --proto_path=$GOPATH/src:#{gogo_dir}/src:. --gogofast_out=$GOPATH/src proto/metrics/agent_payload.proto
+      PATH=#{toolchain_bin_dir} #{protoc_legacy_binary} --proto_path=$GOPATH/src:#{gogo_include}:#{toolchain_legacy_include_dir}:. --gogofast_out=$GOPATH/src proto/metrics/agent_payload.proto
+      echo "done"
 
       echo "Generating metrics proto (python)"
-      #{protoc_legacy_binary} --proto_path=#{gogo_dir}/src:$GOPATH/src:./proto/metrics --python_out=python agent_payload.proto
+      PATH=#{toolchain_bin_dir} #{protoc_legacy_binary} --proto_path=#{toolchain_legacy_include_dir}:#{gogo_include}:./proto/metrics --python_out=python agent_payload.proto
 
       echo "Generating process proto (go)"
-      PATH=/tmp/gogo-bin-v1.3.2 #{protoc_legacy_binary} --proto_path=$GOPATH/src:#{gogo_dir}/src:. --gogofaster_out=$GOPATH/src proto/process/*.proto
+      PATH=#{toolchain_bin_dir} #{protoc_binary} --proto_path=#{toolchain_legacy_include_dir}:#{gogo_include}:. --gogofaster_out=$GOPATH/src proto/process/*.proto
```

Comment:
Is it intended to change from `protoc_legacy_binary` to `protoc_binary` here?

Reply:
It was not, but I think since CI is verifying that the generated code isn't changing it's fine to upgrade (that's an expected outcome since both should just call the gogofaster binary )

Reply:
Hmm - I'm a bit afraid of unintended changes to be honest of which I'm not fully sure of in the broader scope. Maybe this can be split out into a separate PR to ensure 1:1 functionality in this one?